src/util/expr.h

```diff
@@ -319,6 +319,40 @@ class exprt:public irept
   const_unique_depth_iteratort unique_depth_cend() const;
 };
 
+/// Protected class for all expressions.
+/// This API will eventually replace exprt.
+class protected_exprt : public exprt
+{
+protected:
+  // constructors
+  DEPRECATED("use protected_exprt(id, type) instead")
+  protected_exprt() : exprt()
+  {
+  }
+
+  DEPRECATED("use protected_exprt(id, type) instead")
+  explicit protected_exprt(const irep_idt &_id) : exprt(_id)
+  {
+  }
+
+  protected_exprt(const irep_idt &_id, const typet &_type) : exprt(_id, _type)
+  {
+  }
+
+  // protect these low-level methods
+  using exprt::make_bool;
+  using exprt::make_typecast;
+  using exprt::op0;
+  using exprt::op1;
+  using exprt::op2;
+  using exprt::op3;
+  // using exprt::find;
+  // using exprt::get;
+  // using exprt::set;
```

Comment:
Adding commented-out code without comments is just raising questions, but not providing answers.

Reply:
sure, removed